### PR TITLE
[WIP]Add speaker recognition example

### PR DESCRIPTION
**Pull request by KPatr1ck**
## PR types
New feature

## PR changes
dataset, model, example

## Describe
- Model:
  - Add ECAPA-TDNN model
- Datasets:
  - Add VoxCeleb1 dataset
  - Add OpenRIRNoise dataset
  - Update api usage in other datasets
- Data Augment
  - Add on the fly data augment
- Loss
  - Add Additive Angular Margin Loss
- Scripts
  - Add training script of speaker identification
  - Add speaker verification and EER evaluation

### CODE REVIEW

**Comment by ZeyuChen** [comment]
@KPatr1ck any update？

**Comment by KPatr1ck** [comment]
> @KPatr1ck any update？

数据集，模型，loss，训练和预测脚本已齐。
全量数据在训练对齐中。

**Comment by KPatr1ck** [comment]
@ZeyuChen @ranchlai 
Doc string持续补齐中，model、dataset、训练预测脚本可以先reivew一波～

File: PaddleAudio/examples/speaker_recognition/model.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def forward(self, x, lengths=None):
+        # x.shape: (N, C, L)
+        x = self.backbone(x, lengths).squeeze(
+            -1)  # (N, emb_size, 1) -> (N, emb_size)
+        x = self.dropout(x)
+
+        for fc in self.blocks:
+            x = fc(x)
+
+        # KP: W和x的向量归一化，输出为余弦相似度，供Additive Angular Margin计算loss
```
Comment (ZeyuChen) [inline PaddleAudio/examples/speaker_recognition/model.py:58]: 使用标准的注释
`# NOTE(xxxxxgithuid or username): blabla`

File: PaddleAudio/examples/speaker_recognition/signal_processing.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        return paddle.clip(20 * paddle.log10(out), min=-80)
+    else:
+        raise NotImplementedError
+
+
+def dB_to_amplitude(SNR):
+    return 10**(SNR / 20)
+
+
+def convolve1d(
```
Comment (ZeyuChen) [inline PaddleAudio/examples/speaker_recognition/signal_processing.py:61]: convolved1d和conv1d的区别是什么？

File: PaddleAudio/examples/speaker_recognition/signal_processing.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    hhpf = sinc(3 * (notch_freq + notch_width) * inputs)
+    hhpf *= blackman_window(filter_width)
+    hhpf /= -paddle.sum(hhpf)
+    hhpf[pad] += 1
+
+    # Adding filters creates notch filter
+    return (hlpf + hhpf).reshape([1, -1, 1])
+
+
+def reverberate(waveforms,
```
Comment (ZeyuChen) [inline PaddleAudio/examples/speaker_recognition/signal_processing.py:135]: 参考SpeechBrain看这些基础function是否有注释

File: PaddleAudio/examples/speaker_recognition/speaker_verification.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+                ids, waveforms, lengths = batch['ids'], batch[
+                    'waveforms'], batch['lengths']
+                feats = feature_extractor(waveforms)
+                feats = feature_normalize(
+                    feats, lengths, mean_norm=True, std_norm=False)
+                embeddings = model.backbone(feats, lengths).squeeze(-1)
+
+                id2embedding_norm.update(dict(zip(ids, embeddings)))
+
+        # Score normalization based on trainning samples.
```
Comment (ZeyuChen) [inline PaddleAudio/examples/speaker_recognition/speaker_verification.py:233]: trainning -> training

File: PaddleAudio/paddleaudio/datasets/open_rir_noise.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+from ..backends import save_wav
+from ..utils import DATA_HOME, decompress, download_and_decompress, get_logger
+from .dataset import feat_funcs
+
+logger = get_logger()
+
+__all__ = ['OpenRIRNoise']
+
+
+class OpenRIRNoise(Dataset):
```
Comment (ZeyuChen) [inline PaddleAudio/paddleaudio/datasets/open_rir_noise.py:35]: Need more comments for dataset source

File: PaddleAudio/paddleaudio/datasets/voxceleb1.py [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+from ..backends import load as load_audio
+from ..utils import DATA_HOME, decompress, download_and_decompress, get_logger
+from .dataset import feat_funcs
+
+logger = get_logger()
+
+__all__ = ['VoxCeleb1']
+
+
+class VoxCeleb1(Dataset):
```
Comment (ZeyuChen) [inline PaddleAudio/paddleaudio/datasets/voxceleb1.py:34]: More dataset source comments.

File: PaddleAudio/paddleaudio/models/ecapa_tdnn.py [hunk 7]
```diff
@@ -0,0 +1,414 @@
...
+
+        x = self.tdnn1(x)
+        x = self.res2net_block(x)
+        x = self.tdnn2(x)
+        x = self.se_block(x, lengths)
+
+        return x + residual
+
+
+class ECAPA_TDNN(nn.Layer):
```
Comment (ZeyuChen) [inline PaddleAudio/paddleaudio/models/ecapa_tdnn.py:325]: Move this model implementation to examples if without pretrained weight.
